substrate/client/rpc-spec-v2/src/transaction/tests/transaction_tests.rs

```diff
@@ -235,3 +236,48 @@ async fn tx_slow_client_replace_old_messages() {
 
 	assert_eq!(res, exp);
 }
+
+#[tokio::test]
+async fn tx_with_monitoring() {
+	let (api, pool, client, tx_api, _exec_middleware, _pool_middleware, mut tx_monitor) =
+		setup_api_tx_with_monitoring();
+	let block_1_header = api.push_block(1, vec![], true);
+	client.set_best_block(block_1_header.hash(), 1);
+
+	let uxt = uxt(Alice, ALICE_NONCE);
+	let xt = hex_string(&uxt.encode());
+
+	let mut sub = tx_api
+		.subscribe_unbounded("transactionWatch_v1_submitAndWatch", rpc_params![&xt])
+		.await
+		.unwrap();
+
+	let event: TransactionEvent<H256> = get_next_event_sub!(&mut sub);
+	assert_eq!(event, TransactionEvent::Validated);
+
+	// Import block 2 with the transaction included.
+	let block_2_header = api.push_block(2, vec![uxt.clone()], true);
+	let block_2 = block_2_header.hash();
+	let event = ChainEvent::NewBestBlock { hash: block_2, tree_route: None };
+	pool.inner_pool.maintain(event).await;
+
+	let event: TransactionEvent<H256> = get_next_event_sub!(&mut sub);
+	assert_eq!(
+		event,
+		TransactionEvent::BestChainBlockIncluded(Some(TransactionBlock {
+			hash: block_2,
+			index: 0
+		}))
+	);
+
+	// Check the monitor produces the same event
+	let result = tokio::time::timeout(std::time::Duration::from_secs(5), tx_monitor.next())
```

Comment:
nit: not sure how 5 seconds stands against our CI hiccups. Thought that this should be more resilient, as in 10x what we'd expect to wait under normal CI load. If 5s is the number, please ignore.

```suggestion
	let result = tokio::time::timeout(std::time::Duration::from_secs(5), tx_monitor.next())
```